pkg/core/blockchain.go

```diff
@@ -751,6 +758,7 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 	for _, f := range bc.postBlock {
 		f(bc, txpool, block)
 	}
+	bc.updateExtensibleWhitelist(block.Index)
```

Comment:
Can we avoid doing that if there are not relevant changes in the block?

Reply:
We are checking if cache is valid inside this function.

Reply:
It still a little more than I'd like to. And it's in the `storeBlock`, thus affecting every block.

Reply:
Right now it just checks cached values for 3 lists (because they are updated in `PostPersist`). How do you want to optimize this?

Reply:
OK, let's leave it as is and see whether it affects performance really, then change if need be.

Reply:
I have tested this on dump restore (small block), there were no changes in running time.
But only if we cache all roles, that's why first commit is there.